File: Symmentric_Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    //Time O(N)
    //Space O(H)
    boolean isValid;
    public boolean isSymmetric(TreeNode root) {
        if(root == null)
        {
            return true;
        }
        return helper(root.left , root.right);
    }
    public boolean helper(TreeNode Left, TreeNode Right)
    {
        //base
        //Logic
        if(Left == null && Right == null)
        {
            return true;
        }
        if((Left == null || Right == null) ||(Left.val != Right.val))
        {
            return false;
        }
        return helper(Left.left, Right.right) && helper(Left.right, Right.left);
    }
}

File: symmentric_tree.py
# TC : O(N)
# SC : O(N)

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return False

        stack = [(root.left, root.right)]
        while stack:
            left_tree, right_tree = stack.pop()
            if not left_tree and not right_tree:
                continue
            if not left_tree or not right_tree:
                return False
            if left_tree.val != right_tree.val:
                return False
            stack.append((left_tree.right, right_tree.left))
            stack.append((left_tree.left, right_tree.right))
        
        return True

              